Core/src/main/java/com/plotsquared/core/queue/QueueProvider.java

```diff
@@ -25,24 +25,17 @@
  */
 package com.plotsquared.core.queue;
 
+import com.sk89q.worldedit.world.World;
+
+import javax.annotation.Nonnull;
+
 public abstract class QueueProvider {
-    public static QueueProvider of(final Class<? extends LocalBlockQueue> primary,
-        final Class<? extends LocalBlockQueue> fallback) {
+    public static QueueProvider of(@Nonnull final Class<? extends QueueCoordinator> primary) {
         return new QueueProvider() {
 
-            private boolean failed = false;
-
-            @Override public LocalBlockQueue getNewQueue(String world) {
-                if (!failed) {
-                    try {
-                        return (LocalBlockQueue) primary.getConstructors()[0].newInstance(world);
-                    } catch (Throwable e) {
-                        e.printStackTrace();
-                        failed = true;
-                    }
-                }
+            @Override public QueueCoordinator getNewQueue(@Nonnull World world) {
                 try {
-                    return (LocalBlockQueue) fallback.getConstructors()[0].newInstance(world);
+                    return (QueueCoordinator) primary.getConstructors()[0].newInstance(world);
                 } catch (Throwable e) {
                     e.printStackTrace();
```

Comment:
We need better error handling here, and at the very least a more descriptive error (log something)